### DISCUSSION TITLE
chat history.

### DISCUSSION THREAD

**Discussion by EspenMarsteinSandtveit** [aurelio-labs/semantic-router#137] [Q&A]
Hi how are you guys handling follow up questions and chat history?

**Comment by andreped** [comment]
Any comments, @jamescalam and @bruvduroiu?

Does the current `LayerRoute` solution, or `semantic-router` solution in general, only work with single user questions, or is there a mechanism for it to use chat history in the classification?

**Comment by bruvduroiu** [comment]
At the moment, handling chat history and follow up questions are not in scope for functionalities of semantic router. However, you can implement your own chat history handling strategy.

One example might be just concatenating the chat history before passing to `RouteLayer.__call__`.

cc @andreped

**Reply by andreped** [reply]
Hello, @bruvduroiu! Thanks for replying :] We handle the chat history for the rest of our solution, but we wish to classify user questions. For this classification we have used semantic router.

However, we have come to realize that the classifier does not take into account the chat history when performing the classification. There are lots of situations where this would make sense - the question needs to be classified also based on the current context.

We were just wondering if there is a way to do so with semantic router, or if you have any suggestions on how to proceed.
